### PR TITLE
[range.iota,range.adaptors] Add cross-references for private member t…

### LINKED ISSUE #2716: [range.iota, range.adaptors] View's synopses missing cross-references to iterator and sentinel members
Per https://github.com/cplusplus/draft/pull/2475#issuecomment-466289153, these are bugs.

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #2716.

### CODE REVIEW

File: source/ranges.tex [hunk 1]
```diff
@@ -2881,7 +2885,9 @@
              RegularInvocable<F&, iter_reference_t<iterator_t<V>>>
   class transform_view : public view_interface<transform_view<V, F>> {
   private:
+    // \ref{range.transform.iterator}, class \tcode{transform_view::iterator}
     template<bool> struct iterator;             // \expos
+    // \ref{range.transform.sentinel}, class \tcode{transform_view::sentinel}
```
Comment (johelegp) [inline source/ranges.tex]: These are class templates.

File: source/ranges.tex [hunk 2]
```diff
@@ -3560,6 +3566,7 @@
   private:
     V base_ = V();                                      // \expos
     iter_difference_t<iterator_t<V>> count_ = 0;        // \expos
+    // \ref{range.take.sentinel}, class \tcode{take_view::sentinel}
```
Comment (johelegp) [inline source/ranges.tex]: Also a class template.

File: source/ranges.tex [hunk 3]
```diff
@@ -3782,8 +3789,10 @@
   private:
     using InnerRng =                    // \expos
       iter_reference_t<iterator_t<V>>;
+    // \ref{range.join.iterator}, class \tcode{join_view::iterator}
     template<bool Const>
       struct iterator;                  // \expos
+    // \ref{range.join.sentinel}, class \tcode{join_view::sentinel}
```
Comment (johelegp) [inline source/ranges.tex]: Class templates too.

File: source/ranges.tex [hunk 4]
```diff
@@ -4309,7 +4318,9 @@
     V base_ = V();                              // \expos
     Pattern pattern_ = Pattern();               // \expos
     iterator_t<V> current_ = iterator_t<V>();   // \expos, present only if \tcode{!ForwardRange<V>}
+    // \ref{range.split.outer}, class \tcode{split_view::outer_iterator}
     template<bool> struct outer_iterator;       // \expos
+    // \ref{range.split.inner}, class \tcode{split_view::inner_iterator}
```
Comment (johelegp) [inline source/ranges.tex]: Same as above.

File: source/ranges.tex (hunk 1 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex (hunk 2 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex (hunk 3 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex (hunk 4 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

**Comment by CaseyCarter** [comment]
`split_view::outer_iterator::value_type` is another case. It currently has a trailing `// see \ref{range.split.outer.value}` comment, which should be changed to be consistent with the style of this PR.

**Comment by zygoloid** [comment]
Merging now for post-Kona, let's make further fixes in a separate PR.

**Pull request merged**